docs/misc/components.md

```diff
@@ -0,0 +1,118 @@
+
+// Sets style of the text to be blue and italic
+text.setStyle(blueItalic);
+
+// Overwrites blue and italic style to be red, bold, underlined, and strikethrough
+text.withStyle(red).withStyle(bold).withStyle(doubleLines);
+```
+This results in red, bold text with two lines as shown in the image:
+![red_hello]
+
+Chaining Components
+-------------------
+
+`MutableComponent` can have additional components chained as siblings with `MutableComponent#append`. Chained components can be retrieved with `MutableComponent#getSiblings`.
+
+`Component` stores its siblings like a tree and traversed in preorder; the parent style is merged with those of its siblings.
+![tree]
+
+The code below will create a component with the same structure in the above example:
+```java
+// Create text only components
+MutableComponent first = Component.literal("first ");
+MutableComponent second = Component.literal("second ");
+MutableComponent third = Component.literal("third ");
+MutableComponent fourth = Component.literal("fourth ");
+MutableComponent fifth = Component.literal("fifth ");
+MutableComponent sixth = Component.literal("sixth ");
+MutableComponent seventh = Component.literal("seventh ");
+
+// Create components with style
+MutableComponent red = Component.litearl("red ").withStyle(Style.EMPTY.withColor(0xFF0000));
+MutableComponent blue = Component.literal("blue ").withStyle(Style.EMPTY.withColor(0x0000FF));
+MutableComponent bold = Component.literal("bold ").withStyle(Style.EMPTY.withBold(true));
+
+// Structure created components in the same way as the image
+red.append(first).append(blue).append(seventh);
+blue.append(second).append(third).append(bold);
+bold.append(fourth).append(fifth).append(sixth);
+```
+Here's how it looks like in-game
+![style_annotated]
+
+Text Formatting
+---------------
+
+Text formatting is the process of inserting data as text into predefined larger text.
+It can be used for displaying coordinate with x, y, z annotation, showing number with respectful units alongside, etc.
+Usually special notation called **format specifiers** are used for indicating where a text can be inserted into.
+
+`TranslatableContents` supports text formatting and uses two types of format specifiers: `%s` and `%n$s`.
+It has object array `args` for holding what to insert in place of format specifiers.
+
+`%s` is replaced with elements of `args` in order they appear, i.e., First `%s` is replaced with a first element of `args`, and so on.
+`%n$s` is positional specifier; they can specify which element will replace them with number `n`.
+* Formatting `x:%s y:%s z:%s` with `[1, 2, 3]` as `args` results in `x:1 y:2 z:3`
+* Formatting `Time: %1$s ms` with `17` as `args` results in `Time: 17 ms`
+* Formatting `Player name: %2$s, HP: %1$s` with `[10.2, Dev]` as `args` results in `Player name: Dev, HP: 10.2`
+
+`args`'s elements may be `Component`s that will be inserted into the resulting formatted text with all their attributes preserved.
```

Comment:
```suggestion
When an element within `args` is a `Component`, the component will be formatted as normal.
```

Reply:
I don't think this sufficiently conveys how the component will be formatted, what does it mean by "normal"?

Reply:
I was thinking as normal or as specified since we already defined how a component is formatted above.

Reply:
That's a fair point on the "normal" sentence. Hmm, it still feels a bit clunky to read though.

What about this?
```
Any `Component` element within `args` will be transformed into a formatted text string.
```